core/kazoo_json/src/kz_json.erl

```diff
@@ -402,6 +406,22 @@ sum(?JSON_WRAPPER(_)=JObj1, Value, Sumer, Keys)
     V = get_value(Syek, JObj1),
     set_value(Syek, Sumer(V, Value), JObj1).
 
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Reorder JSON objects according to the given list of binaries.
+%% Given Path MUST resolve to all distinct values in the given objects.
+%% These resolved values MUST all be in the list of binaries too.
+%% List of binaries MUST NOT contain duplicates.
+%% Both lists MUST be of same size.
+%% @end
+%%--------------------------------------------------------------------
+-spec order_by(path(), ne_binaries(), objects()) -> objects().
+order_by(Path, Ids, JObjs)
```

Comment:
@lazedo @fenollp In this case, it seems like a specific feature to kz_datamgr and not a "generic" JSON object operation. I would put it there and maybe name it `sort_by_ids/3` maybe? As always, naming is hard.